### PR TITLE
Js hangout maps

### PR DESCRIPTION
**Pull request by Kirazuto7**
Changes:
Images: Added icons for travel options

UserService.js: Used to parse user data from firebase, particularly the userlocation at the moment

Index.js: Removed Strict mode to stop useEffect from rendering twice

App.js: pass parsed user location to the card homepage 

CardDeck: search for hangouts based on user location and pass down coordinates of the location as props, change the shuffleDeck function to return a copy of the array rather than the original array

Map.js: renders a map using the google maps api with markers for the hangout and user location. fetches directions from the api if a travel mode is set

HangoutDetail.js: Insert google maps in the details container and added buttons to set the travel mode of the mode if the user wants to see travel directions. The time to travel and steps will be displayed when directions are requested. Otherwise markers are shown displaying the locations of the user and hangout location.

### CODE REVIEW

**Review by JChen0915 — APPROVED** [review]
A lot of changes for one pull request but the addition of these features make use of preexisting user data, seems good.

File: my-app/src/Components/App.js
Comment (chizy7) [inline my-app/src/Components/App.js:1]: The changes in the App.js file are clear and well-structured.

File: my-app/src/Components/Card.js
Comment (chizy7) [inline my-app/src/Components/Card.js:1]: The changes in the Card.js file are clear and relevant to the functionality being added.

File: my-app/src/Components/CardDeck.js
Comment (chizy7) [inline my-app/src/Components/CardDeck.js:1]: The changes in the CardDeck.js file are clear, relevant, and correctly implemented.

File: my-app/src/Components/HangoutDetail.js
Comment (chizy7) [inline my-app/src/Components/HangoutDetail.js:1]: The changes in the HangoutDetail.js file are clear, relevant, and correctly implemented.

File: my-app/src/Components/HomepageWithCards.js
Comment (chizy7) [inline my-app/src/Components/HomepageWithCards.js:1]: The changes in the HomepageWithCards.js file are clear, relevant, and correctly implemented.

File: my-app/src/Components/Map.js
Comment (chizy7) [inline my-app/src/Components/Map.js:1]: This looks good too!

File: my-app/src/Services/UserService.js
Comment (chizy7) [inline my-app/src/Services/UserService.js:1]: Looks good as well.

**Review by chizy7 — APPROVED** [review]
Overall all looks good and properly implemented.

File: my-app/src/Services/UserService.js
Comment (beaverseven) [reply my-app/src/Services/UserService.js:1]: Agreed.

**Comment by beaverseven** [comment]
The implementation of google maps into our hangoutDetails looks great!

**Review by beaverseven — APPROVED** [review]
A lot of changes in a single PR, maybe might want to split it up in the near future. But a lot of implementation has been added, which is great.

**Pull request merged**